docs/public-api-for-1.0.md

```diff
@@ -0,0 +1,711 @@
+support configuring the behavior according to the OTel specification.
+
+Therefore, we propose to make all the existing API not expressly discussed in the above bootstrap, backends, and
+configuration sections, internal.
+
+Reducing the API surface to just that which is required to meet these goals paves an easier path for a 1.0 release, and
+does not rule out expanding it the future to support deeper customization.
+
+Where extensibility is required, adopters should compose the opaque backends with their own logic. As a concrete
+example, if an adopter wishes to pre-sample traces in a way that is not supported in the OTel specification, they should
+construct their own wrapper type that conforms to `Tracing.Instrument`, which wraps the backend returned by
+`OTel.makeTracingBackend`.
+
+### Package structure and traits
+
+Swift OTel will provide a single library product, `OTel`, with the public API discussed above.
+
+Any other existing library products will become internal targets. Concretely, the majority of the existing `OTel`
+library product will move to an internal `OTelCore` target; and `OTLPGRPC` will be made an internal detail.
+
+Support for OTLP/gRPC and OTLP/HTTP will be gated behind package traits: `OTLPGRPC` and `OTLPHTTP`, respectively, which
+will both be enabled by default. This allows adopters to only include the dependencies for a single exporter.
+
+Swift OTel might opt to make the corresponding configuration values only reachable in code with the trait enabled, e.g.:
+
+```swift
+var config = OTel.Configuration.default
+config.traces.otlp.protocol = .grpc  // compiler error: Using the OTLP/GRPC exporter requires the `OTLPGRPC` trait enabled.
+```
+
+Attempting to use these configuration values using their corresponding environment variable values will result in a
+runtime error, e.g.:
+
+```swift
+// Assuming OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc, then...
+OTel.bootstrap()  // runtime error: Using the OTLP/GRPC exporter requires the `OTLPGRPC` trait enabled.
+```
+
+### Logging within Swift OTel
+
+There has been some historic confusion related to logging within the Swift OTel library itself, which stems from:
+
+1. Swift Log's process-wide, one-shot bootstrap mechanism for logging; and
+2. Swift OTel providing an OTLP logging backend.
+
+At first glance this presents a chicken-and-egg problem: Do Swift OTel's own logs go to OTLP? What about pre-bootstrap
+logging? Where do errors about exporting logs, get logged? Or the handling of unsupported configuration values?
+
+These concerns are addressed by the OTel spec, which states that such "self-diagnostics" can be handled with any
+language-specific conventions and/or callbacks[^2].
+
+Swift OTel will handle this by having it's own `Logger`, which it will construct without the bootstrapped logging
+backend and will log such diagnostics to standard error.
+
+Future implementations may support customizing the internal logger used by the SDK.
```

Comment:
This is no longer a *future* implementation since you incorporated the feedback into the 1.0 proposal, right?